pysat/_instrument.py

```diff
@@ -441,7 +441,7 @@ def __getitem_xarray__(self, key):
                 # support slicing time, variable name
                 try:
                     return self.data.isel(time=key[0])[key[1]]
-                except:
+                except (TypeError, KeyError):
```

Comment:
Rationale for using try / except here rather than if / else: the data structure have a number of different types of keys and combinations of keys.  From the example code in this file:
```
            # By name
            inst['name']
            # By position
            inst[row_index, 'name']
            # Slicing by row
            inst[row1:row2, 'name']
            # By Date
            inst[datetime, 'name']
            # Slicing by date, inclusive
            inst[datetime1:datetime2, 'name']
            # Slicing by name and row/date
            inst[datetime1:datetime2, 'name1':'name2']
```